### PR TITLE
LayoutRendererWrappers - Almost always support AppDomainFixedOutput

### PR DESCRIPTION
**Pull request by snakefoot**
When Inner-Layout allows it.

### CODE REVIEW

File: src/NLog/LayoutRenderers/MachineNameLayoutRenderer.cs [hunk 1]
```diff
@@ -48,7 +48,7 @@ namespace NLog.LayoutRenderers
     [AppDomainFixedOutput]
     [ThreadAgnostic]
     [ThreadSafe]
-    public class MachineNameLayoutRenderer : LayoutRenderer, IStringValueRenderer
```
Comment (304NotModified) [inline src/NLog/LayoutRenderers/MachineNameLayoutRenderer.cs]: this is this removed? I don't understand in the context of this PR

File: src/NLog/LayoutRenderers/ThreadNameLayoutRenderer.cs [hunk 2]
```diff
@@ -59,6 +59,6 @@ private static string GetStringValue()
 #endif
         }
 
-        string IStringValueRenderer.GetFormattedString(LogEventInfo _) => GetStringValue();
```
Comment (304NotModified) [inline src/NLog/LayoutRenderers/ThreadNameLayoutRenderer.cs]: `_` is fine here?

File: src/NLog/LayoutRenderers/MachineNameLayoutRenderer.cs (hunk 1 above)
Comment (snakefoot) [reply src/NLog/LayoutRenderers/MachineNameLayoutRenderer.cs]: There is no need for IStringValueRenderer when having the attribute `[AppDomainFixedOutput]`

File: src/NLog/LayoutRenderers/ThreadNameLayoutRenderer.cs (hunk 2 above)
Comment (snakefoot) [reply src/NLog/LayoutRenderers/ThreadNameLayoutRenderer.cs]: I know. But it is reported as warning by Sonar: 

https://sonarcloud.io/project/issues?branch=dev&id=nlog&open=AWb1jYMNW3Y69bz5TTB3&resolved=false

File: src/NLog/LayoutRenderers/MachineNameLayoutRenderer.cs (hunk 1 above)
Comment (snakefoot) [reply src/NLog/LayoutRenderers/MachineNameLayoutRenderer.cs]: Have added unit-tests to verify compatibility between internal interfaces and attributes.

File: src/NLog/LayoutRenderers/ThreadNameLayoutRenderer.cs (hunk 2 above)
Comment (304NotModified) [reply src/NLog/LayoutRenderers/ThreadNameLayoutRenderer.cs]: mmm disagree on that. Unfortunately, I cannot set the rules in sonar for NLog now (need to move the repo to a new org)

File: src/NLog/LayoutRenderers/MachineNameLayoutRenderer.cs (hunk 1 above)
Comment (304NotModified) [reply src/NLog/LayoutRenderers/MachineNameLayoutRenderer.cs]: great!

**Pull request merged**